python/ray/train/base_trainer.py

```diff
@@ -336,6 +332,8 @@ def as_trainable(self) -> Type[Trainable]:
         def train_func(config, checkpoint_dir=None):
             # config already contains merged values.
             # Instantiate new Trainer in Trainable.
+            if isinstance(config.get("scaling_config"), dict):
+                config["scaling_config"] = ScalingConfig(**config["scaling_config"])
```

Comment:
Is it possible for this to be the dataclass (i.e. do this conversion in Tune)? Then the Trainer is only aware of the dataclass, and Tune handles both dataclass -> dict and dict -> dataclass conversions.